rust/src/client/mobile_client/mod.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{client::Proxy, mask::model::Model};
+use std::{cell::RefCell, rc::Rc};
+
+mod client;
+use self::client::ClientStateMachine;
+
+pub mod participant;
+use self::participant::ParticipantSettings;
+
+pub struct MobileClient {
+    runtime: tokio::runtime::Runtime,
+    local_model: Rc<RefCell<Option<Model>>>,
+    global_model: Rc<RefCell<Option<Model>>>,
+    client_state: Option<ClientStateMachine>,
+}
+
+impl MobileClient {
+    pub fn new(url: &str, participant_settings: ParticipantSettings) -> Self {
+        let runtime = tokio::runtime::Builder::new()
+            .basic_scheduler()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let local_model = Rc::new(RefCell::new(None));
+        let global_model = Rc::new(RefCell::new(None));
+
+        let client_state = ClientStateMachine::new(
+            Proxy::new_remote(url),
+            participant_settings,
+            local_model.clone(),
+            global_model.clone(),
+        )
+        .unwrap();
+
+        Self {
+            runtime,
+            local_model,
+            global_model,
+            client_state: Some(client_state),
+        }
+    }
+
+    pub fn set_local_model(&self, local_model: Model) {
+        *self.local_model.borrow_mut() = Some(local_model);
+    }
+
+    pub fn get_global_model(&self) -> Option<Model> {
+        self.global_model.borrow().clone()
+    }
```

Comment:
we should either have a `# Panics` section in those method docs or switch to `try_borrow_mut()`/`try_borrow()` and add some error handling.